src/cache/fetchWithCache.ts

```diff
@@ -0,0 +1,26 @@
+import { getCache, setCache, delCache } from './cache';
+
+const fetchWithCache = async (url: string): Promise<unknown> => {
+  const cachedResponse = getCache(url);
+
+  if (cachedResponse) {
+    console.log('Returning cached response');
```

Comment:
here it is necessary to create a log error messages, so in the future we can take off with a graylog tools or similar tools